chain/actors/builtin/miner/miner.go

```diff
@@ -0,0 +1,113 @@
+package miner
+
+import (
+	"github.com/filecoin-project/go-state-types/dline"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-bitfield"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/cbor"
+	v0builtin "github.com/filecoin-project/specs-actors/actors/builtin"
+	v0miner "github.com/filecoin-project/specs-actors/actors/builtin/miner"
+
+	"github.com/filecoin-project/lotus/chain/actors/adt"
+	"github.com/filecoin-project/lotus/chain/types"
+)
+
+var Address = v0builtin.InitActorAddr
+
+func Load(store adt.Store, act *types.Actor) (st State, err error) {
+	switch act.Code {
+	case v0builtin.StorageMinerActorCodeID:
+		out := v0State{store: store}
+		err := store.Get(store.Context(), act.Head, &out)
+		if err != nil {
+			return nil, err
+		}
+		return &out, nil
+	}
+	return nil, xerrors.Errorf("unknown actor code %s", act.Code)
+}
+
+type State interface {
+	cbor.Marshaler
+
+	AvailableBalance(abi.TokenAmount) (abi.TokenAmount, error)
+	VestedFunds(abi.ChainEpoch) (abi.TokenAmount, error)
+
+	GetSector(abi.SectorNumber) (*SectorOnChainInfo, error)
+	FindSector(abi.SectorNumber) (*SectorLocation, error)
+	GetSectorExpiration(abi.SectorNumber) (*SectorExpiration, error)
+	GetPrecommittedSector(abi.SectorNumber) (*SectorPreCommitOnChainInfo, error)
+	LoadSectorsFromSet(filter *bitfield.BitField, filterOut bool) (adt.Array, error)
```

Comment:
We should instead introduce a `LoadSectorsForWinningPoSt` and `LoadSectorsForWindowPoSt` function.

Reply:
Ok, I've changed my mind. We need to load sectors for _future_ proofs. That means we need future logic, not the current logic.